tests/ci/integration/run_socat_integration.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash -exu
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+source tests/ci/common_posix_setup.sh
+
+# Set up environment.
+# SRC_ROOT(aws-lc)
+#  - SCRATCH_FOLDER
+#    - SOCAT_SRC
+#    - AWS_LC_BUILD_FOLDER
+#    - AWS_LC_INSTALL_FOLDER
+
+# Assumes script is executed from the root of aws-lc directory
+SCRATCH_FOLDER=${SRC_ROOT}/"scratch"
+AWS_LC_BUILD_FOLDER="${SCRATCH_FOLDER}/aws-lc-build"
+AWS_LC_INSTALL_FOLDER="${SCRATCH_FOLDER}/aws-lc-install"
+SOCAT_SRC="${SCRATCH_FOLDER}/socat"
+
+function build_and_test_socat() {
+  pushd "$SOCAT_SRC"
+  autoconf
+  ./configure --enable-openssl-base="$AWS_LC_INSTALL_FOLDER"
+  make -j "$NUM_CPU_THREADS"
+  # test 146 OPENSSLLISTENDSA: fails because AWS-LC doesn't support FFDH ciphersuites which are needed for DSA
+  # test 216 UDP6MULTICAST_UNIDIR: known flaky test in socat with newer kernels
+  # test 309 OPENSSLRENEG1: AWS-LC doesn't support renegotiation by default, it can be enabled by calling SSL_set_renegotiate_mode
+  # but that has caveats. See PORTING.md 'TLS renegotiation'
+  # test 310 OPENSSLRENEG2: AWS-LC doesn't support renegotiation by default, it can be enabled by calling SSL_set_renegotiate_mode
+  # but that has caveats. See PORTING.md 'TLS renegotiation'
+  # test 399 OPENSSL_DTLS_CLIENT: Unknown issue running openssl s_server
```

Comment:
Is this using the system's `openssl` executable? Is this using something we should support

Reply:
Eventually we will need to support it, but right now the `openssl` CLI and our `bssl` CLI are completely incompatible (different names, options, supported options...)

Reply:
I can't figure out what the issue is or replicate it locally:
```
test 399 OPENSSL_DTLS_CLIENT: OpenSSL DTLS client... FAILED:  ./socat:
 openssl s_server -4 -dtls -accept 12212 -quiet   -cert testsrv.pem &
 ./socat -t 0.1  -T 3 - OPENSSL-DTLS-CLIENT:localhost:12212,pf=ip4,verify=0,
1d0
```
https://github.com/aws/aws-lc/actions/runs/7617226582/job/20745774924?pr=1387#step:4:10143 and that's with both `-d` debug and `-v` verbose enabled.

Reply:
Ah yeah, just double checking. I agree that the `openssl` CLI will have to be something we start from scratch.